R/TileDBArray.R

```diff
@@ -853,9 +853,11 @@ setMethod("[", "tiledb_array",
                          MoreArgs=list(qryptr=qryptr), SIMPLIFY=TRUE)
       ## ensure > 0 for correct handling of zero-length outputs, ensure respecting memory budget
       spdl::debug("['['] result of size estimates is {}", paste(ressizes, collapse=","))
-      resrv <- max(1, min(memory_budget/8, ressizes[ressizes > 0]))
+      .idx_lgl <- ressizes > 0; ressizes <- if (any(.idx_lgl)) ressizes[ressizes > 0] else 0
+      resrv <- max(1, min(memory_budget/8, ressizes))
```

Comment:
Minor style fix to put separate statements on individual lines; also re-use the index generated by `resizes > 0` to avoid re-running the comparison (in R, we can use logicals to index a vector where `TRUE` means "take" and `FALSE` means "drop")
```suggestion
      idx <- ressizes > 0
      ressizes <- if (any(idx)) {
        ressizes[idx] 
      } else {
        0
      }
      resrv <- max(1, min(memory_budget/8, ressizes))
```

Reply:
✨